fluent/client/errors.go

```diff
@@ -0,0 +1,45 @@
+package client
+
+import (
+	"fmt"
+	"net/http"
+)
+
+type WSConnError struct {
+	StatusCode   int
+	ResponseBody string
+	ConnErr      error
+}
+
+var nonRetryableStatusCodes = []int{
+	http.StatusBadRequest,
+	http.StatusUnauthorized,
+	http.StatusForbidden,
+	http.StatusNotFound,
+	http.StatusMethodNotAllowed,
+	http.StatusNotImplemented,
+	http.StatusHTTPVersionNotSupported,
+}
+
+func (e *WSConnError) Error() string {
+	if e.ConnErr != nil {
+		return fmt.Sprintf("Connection Error %s: %d. %s", e.ConnErr.Error(), e.StatusCode, e.ResponseBody)
+	}
+
+	return fmt.Sprintf("Connection Error %d: %s", e.StatusCode, e.ResponseBody)
+}
+
+// IsNonRetryableStatusCode checks if the provided HTTP status code is non-retryable
+func (e *WSConnError) IsNonRetryableStatusCode(statusCode int) bool {
+	for _, nonRetryableCode := range nonRetryableStatusCodes {
+		if nonRetryableCode == statusCode {
+			return true
+		}
+	}
+
+	return false
+}
+
+func NewWSConnError(err error, statusCode int, respBody string) *WSConnError {
+	return &WSConnError{ConnErr: err, StatusCode: statusCode, ResponseBody: respBody}
```

Comment:
You could add logic that decides if an error is retryable or not here. You can extend `WsConnError` by adding `bool` property like so:

```
type WSConnError struct {
	StatusCode   int
	ResponseBody string
	ConnErr      error
        retryable bool
}
``` 

And then your IsRetryable method looks like this:

```
func (e *WSConnError) IsRetryable() bool {
    return e.retryable
}
```

Reply:
@mirko-lazarevic I have added 
```bash 
type WSConnError struct {
	StatusCode   int
	ResponseBody string
	ConnErr      error
        retryable bool
}
```

and made change to `NewWSConnError()` with the logic that decides if an error is retryable or not.

But i didn't find the use of since we already have value stored in `WSConnError`
```bash
func (e *WSConnError) IsRetryable() bool {
    return e.retryable
}
```

Reply:
@mirko-lazarevic updated this change